### PR TITLE
[HIPIFY][Tensor][feature] Introduce `cuTensor` support - Part 1 - w/o hipify-perl and documentation

### PR DESCRIPTION
**Pull request by aakanksha555**
+ Add support for tensorStatus
+ Add relevant synthetic test for tensorStatus

### CODE REVIEW

**Comment by emankov** [comment]
It is non-buildable...

**Review by emankov — CHANGES_REQUESTED** [review]
Please, make it buildable, first of all.

**Comment by emankov** [comment]
What about hipify-perl?

**Review by emankov — CHANGES_REQUESTED** [review]
hipify-perl is an integral part of each HIPIFY commit where there are any changes in the API.

**Comment by aakanksha555** [comment]
> hipify-perl is an integral part of each HIPIFY commit where there are any changes in the API.

Got it!

**Comment by aakanksha555** [comment]
@emankov would the tensor support in hipify-perl go into experimentalSubstitutions instead?

File: bin/hipify-perl [hunk 1]
```diff
@@ -6159,6 +6159,7 @@ sub simpleSubstitutions {
     subst("cuda.h", "hip\/hip_runtime.h", "include_cuda_main_header");
     subst("cuda_runtime.h", "hip\/hip_runtime.h", "include_cuda_main_header");
     subst("cudnn.h", "hipDNN.h", "include_cuda_main_header");
+    subst("cutensor.h", "hiptensor.h", "include");
```
Comment (aakanksha555) [inline bin/hipify-perl]: Is include the correct choice here or is it include_cuda_header?

Comment (emankov) [reply bin/hipify-perl]: Nope, as `hipify-perl` is an entirely autogenerated script (`hipify-clang --perl`): https://github.com/ROCm/HIPIFY/blob/amd-staging/bin/hipify-perl#L25.
You should change its generation, not the script itself.

**Comment by emankov** [comment]
> @emankov would the tensor support in hipify-perl go into experimentalSubstitutions instead?

Why? We are going to support all APIs, including those that appeared in previous ROCm HIP versions. `HIP_EXPERIMENTAL` APIs are only those that appear in the latest ROCm HIP release (6.4.0 in our case).

**Comment by emankov** [comment]
@aakanksha555, Could you please revert your last commit with manual changing hipify-perl? I'm going to merge this PR w/o hipify-perl and documentation support. Thanks!

**Comment by aakanksha555** [comment]
> > @emankov would the tensor support in hipify-perl go into experimentalSubstitutions instead?
> 
> Why? We are going to support all APIs, including those that appeared in previous ROCm HIP versions. `HIP_EXPERIMENTAL` APIs are only those that appear in the latest ROCm HIP release (6.4.0 in our case).

Thanks got it. Ya I just wanted to understand the difference.

**Comment by aakanksha555** [comment]
> @aakanksha555, Could you please revert your last commit with manual changing hipify-perl? I'm going to merge this PR w/o hipify-perl and documentation support. Thanks!

Okay sure. I'll revert it.

**Review by emankov — APPROVED** [review]
LGTM, thanks!

**Pull request merged**